### ISSUE TITLE
[log-shipper] Rate limit improvements (QoS)

### ISSUE THREAD

**Issue by nabokihms** [deckhouse/deckhouse#3994]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

There is a [rate limit feature](https://deckhouse.io/documentation/v1/modules/460-log-shipper/cr.html#clusterlogdestination-v1alpha1-spec-ratelimit) that can help users to throttle requests to the destination.  It can prevent overloads. However, it is not convenient for QoS. If a namespace produces more logs than it is supposed to, other namespaces will be limited alongside.

### Proposed Solution

Add the [`key_field`](https://vector.dev/docs/reference/configuration/transforms/throttle/#key_field) to be able to limit requests by namespace or other metadata info, e.g., by controller.

**Comment by dezzzm** [comment]
Thank you. Can you also add exclude.type and exclude.source? https://vector.dev/docs/reference/configuration/transforms/throttle/#exclude
This is necessary to exclude system namespaces (d8-* and kube-*) from throttling.

**Comment by nabokihms** [comment]
Yeah, I think we can add excludes, but exposing the VRL interface is impossible. The vector team does not guarantee compatibility of VRL expressions between different versions of Vector. Instead, we can add a structural config like the [`logFilter`](https://deckhouse.io/documentation/v1/modules/460-log-shipper/cr.html#clusterloggingconfig-v1alpha1-spec-logfilter) option.

**Comment by nabokihms** [comment]
As a workaround now, creating a destination for each namespace is possible.
It is partly related to the issue https://github.com/deckhouse/deckhouse/issues/3104
